vernac/mltop.ml

```diff
@@ -309,6 +309,9 @@ type ml_module_object = {
 }
 
 let add_module_digest m =
+  if not has_dynlink then
+    m, NoDigest
+  else
```

Comment:
Out of curiosity @herbelin, why is this needed?

Reply:
Because otherwise `file_of_name` fails with a user error at `Declare ML Module` time if there is effectively no plugin.

Another solution would be to add `UserError` in the `try with`.

Reply:
I see, thanks for the explanation; I was planning to do a pass in mltop [in particular to add support for dependency loading of plugins, a long requested feature]

I will keep this comment in mind when doing the pass; of course if you are inclined to add the comment on the code itself that would be nice.